examples/text_labels_and_annotations/custom_legends.py

```diff
@@ -0,0 +1,60 @@
+"""
+========================
+Composing Custom Legends
+========================
+
+Composing custom legends piece-by-piece.
+
+Sometimes you don't want a legend that is explicitly tied to data that
+you have plotted. For example, say you have plotted 10 lines, but don't
+want a legend item to show up for each one. If you simply plot the lines
+and call ``ax.legend()``, you will get the following:
+"""
+# sphinx_gallery_thumbnail_number 2
+from matplotlib import rcParams, cycler
+import matplotlib.pyplot as plt
+import numpy as np
+
+N = 10
+data = [np.logspace(0, 1, 100) + np.random.randn(100) + ii for ii in range(N)]
+data = np.array(data).T
+cmap = plt.cm.coolwarm
+rcParams['axes.prop_cycle'] = cycler(color=cmap(np.linspace(0, 1, N)))
+
+fig, ax = plt.subplots()
+lines = ax.plot(data)
+ax.legend(lines)
+
+##############################################################################
+# In this case, we can compose a legend using Matplotlib objects that aren't
+# explicitly tied to the data that was plotted. For example:
+
+from matplotlib.lines import Line2D
+custom_lines = [Line2D([0], [0], color='b', lw=4),
+                Line2D([0], [0], color='grey', lw=4),
+                Line2D([0], [0], color='r', lw=4)]
+
+fig, ax = plt.subplots()
+lines = ax.plot(data)
+ax.legend(custom_lines, ['Cold', 'Medium', 'Hot'])
+
+
+###############################################################################
+# There are many other Matplotlib objects that can be used in this way. Below
+# is a list of common ones.
+
+from matplotlib.patches import Patch
+from matplotlib.lines import Line2D
+
+legend_elements = [Line2D([0], [0], color='b', lw=4, label='Line'),
+                   Line2D([0], [0], marker='o', color='w', label='Scatter',
+                          markerfacecolor='g', markersize=15),
+                   Patch(facecolor='orange', edgecolor='r',
+                         label='Color Patch')]
+
+# Create the figure
+# If you use `ax.legend`, you'll also need to give the labels manually.
+fig, ax = plt.subplots()
+ax.legend(handles=legend_elements, loc=(0.5, 0.5))
```

Comment:
I understand that one kind of gave up on the idea on using out-of-axes legend before the aforementioned @jklymak's PRs are merged, but then is there a peculiar reason not to simply use `loc="center"` instead of `loc=(0.5, 0.5)`. Except of course if one *really* want to align the *lower left* of the legend with the center of the axes.

Reply:
nope, I'm just being dumb :-)